frontend/src/Chat.tsx

```diff
@@ -1,7 +1,38 @@
 import * as React from 'react';
 import classNames from 'classnames';
+import {IGameMode} from './types';
 
-export type Message = {from: string; message: string; from_game?: boolean};
+export type MessageVariant =
+  | 'ResettingGame'
```

Comment:
Are there any ways to get these to serialize to something like `{ResettingGame: true}` instead?

Reply:
there are `serde_json` configuration flags that allow this, but I think it is better to be consistent with the other enums on the wire now, and then change all of them at once.

https://serde.rs/enum-representations.html

internal or external tagging is probably the most JSON-y approach here?